util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py

```diff
@@ -255,3 +256,43 @@ def _parse_url_query(url: str):
     path = parsed_url.path
     query_params = parsed_url.query
     return path, query_params
+
+def redact_query_parameters(url: str) -> str:
+    """Given a string url, redact sensitive query parameter values"""
+    try:
+        parsed = urlparse(url)
+        if not parsed.query:  # No query parameters to redact
+            return url
+            
+        # Check if any of the sensitive parameters are in the query
+        has_sensitive_params = any(param + "=" in parsed.query for param in PARAMS_TO_REDACT)
+        if not has_sensitive_params:
+            return url
+            
+        # Process query parameters
+        query_parts: list[str] = []
+        for query_part in parsed.query.split("&"):
+            if "=" in query_part:
+                param_name, _ = query_part.split("=", 1) # Parameter name and value
+                if param_name in PARAMS_TO_REDACT:
+                    query_parts.append(f"{param_name}=REDACTED")
+                else:
+                    query_parts.append(query_part)
+            else:
+                query_parts.append(query_part)  # Handle params with no value
+                
+        # Reconstruct the URL with redacted query parameters
+        redacted_query = "&".join(query_parts)
+        return urlunparse(
+            (
+                parsed.scheme,
+                parsed.netloc,
+                parsed.path,
+                parsed.params,
+                redacted_query,
+                parsed.fragment,
+            )
+        )
+    except ValueError:  # an unparsable url was passed
```

Comment:
If this is just for urlparse, maybe do an early return instead so we save one indentation level?